ts/components/conversation/TimelineMessage.tsx

```diff
@@ -237,16 +252,24 @@ export function TimelineMessage(props: Props): JSX.Element {
   const shouldShowAdditional =
     doesMessageBodyOverflow(text || '') || !isWindowWidthNotNarrow;
 
-  const multipleAttachments = attachments && attachments.length > 1;
-  const firstAttachment = attachments && attachments[0];
 
+  // TODO This will only get executed on the first time the component is rendered
+  // How to check if attachment[x].pending changes? 
+  const hasPendingAttachments = (): boolean => {
+    // Ensure that no attachment is still pending
+    if (attachments && attachments.length) {
+      return attachments.every(attachment => !attachment.pending);
+    }
+    
+    return false;
+  }
```

Comment:
I could need some help here. 
I added this function to check if any of the attachments is in pending state, and not just the first one, as it was before (maybe that's not even necessary?).
Right now, I think this function is only run once, when the component is rendered, which mean that if the `pending`  state changes later, the result will not change and thus the download button will not be rendered.

How can I make sure that this function gets re-run whenever the `pending` attribute of an attachment changes?
I think using useCallback() or useMemo() is not the right approach here, because it only notices when the reference to `attachments` changes, right?

Reply:
Why is it a function at all? Can't it just be a local variable? Also, the logic seems wrong - the variable is `hasPendingAttachments` but it was checking to ensure that no attachments were pending.

```suggestion
  const hasPendingAttachments =
    attachments?.length && 
    attachments.some(attachment => attachment.pending);
```

Reply:
You're right, the logic was the wrong way round. Thanks for the hint with the `attachments.some()` function. Didn't know this one yet.